internal/core/graphics/image/cache.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+/*!
+This module contains image and caching related types for the run-time library.
+*/
+
+use std::collections::HashMap;
+
+use super::{Image, ImageCacheKey, ImageInner, SharedImageBuffer, SharedPixelBuffer};
+use crate::{slice::Slice, SharedString};
+
+// Cache used to avoid repeatedly decoding images from disk.
+#[derive(Default)]
+pub(crate) struct ImageCache(HashMap<ImageCacheKey, ImageInner>);
+
+thread_local!(pub(crate) static IMAGE_CACHE: core::cell::RefCell<ImageCache>  = Default::default());
```

Comment:
And yet another global data :-(

Is this also not unbounded?
Should it be within a Mutex? (i believe the image itself should be Send, is it not?)

Reply:
This doesn’t add global data. It moves from the GL backend to here.

The qt backed had no such cache, but it also reads the image from disk every time size() is called.

As discussed in the call today, this is where a bounded associative data structure would be a great fit. 

Yes, this could be process global instead of thread local. Not sure it’s worth it at the moment though?